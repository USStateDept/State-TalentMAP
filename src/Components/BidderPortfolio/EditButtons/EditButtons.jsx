import { Component } from 'react';
import PropTypes from 'prop-types';
import { EMPTY_FUNCTION } from '../../../Constants/PropTypes';

class EditButtons extends Component {
  constructor(props) {
    super(props);
    this.state = {
      showSave: this.props.initialShowSave,
    };
  }

  onShow = () => {
    this.setState({ showSave: true }, () => this.props.onChange({ show: true, type: 'show' }));
  };

  onCancel = () => {
    this.setState({ showSave: false }, () => this.props.onChange({ show: false, type: 'cancel' }));
  };

  onSave = () => {
    this.setState({ showSave: false }, () => this.props.onChange({ show: false, type: 'save' }));
  };

  changeSaveState(val) {
    this.setState({ showSave: val });
  }
  render() {
    const { showSave } = this.state;
    return (
      <div className="edit-buttons-container">
        {!showSave && <button onClick={this.onShow}>Edit</button>}
        {showSave && <button onClick={this.onCancel} className="usa-button-secondary">{'Don\'t Save'}</button>}
        {showSave && <button onClick={this.onSave}>Save</button>}
      </div>
    );
  }
}

EditButtons.propTypes = {
  onChange: PropTypes.func,
  initialShowSave: PropTypes.bool,
};

EditButtons.defaultProps = {
  onChange: EMPTY_FUNCTION,
  initialShowSave: false,
};

export default EditButtons;
