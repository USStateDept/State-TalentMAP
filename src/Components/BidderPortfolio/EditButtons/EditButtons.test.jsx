import { shallow } from 'enzyme';
import toJSON from 'enzyme-to-json';
import EditButtons from './EditButtons';

describe('SearchResultsExportLink', () => {
  it('is defined', () => {
    const wrapper = shallow(<EditButtons />);
    expect(wrapper).toBeDefined();
  });

  // [button index, expected output, should call changeSaveState before test]
  [
    [0, { show: true, type: 'show' }, false],
    [0, { show: false, type: 'cancel' }, true],
    [1, { show: false, type: 'save' }, true],
  ].map(m => (
    it(`responds with the correct prop "${m[1].type}" when button index ${m[0]} is clicked`, () => {
      let val;
      function onChange(v) { val = v; }
      const wrapper = shallow(<EditButtons onChange={onChange} showSave={m[2]} />);
      if (m[2]) { wrapper.instance().changeSaveState(true); }
      wrapper.find('button').at(m[0]).simulate('click');
      expect(val).toEqual(m[1]);
    })
  ));

  it('matches snapshot', () => {
    const wrapper = shallow(<EditButtons />);
    expect(toJSON(wrapper)).toMatchSnapshot();
  });
});
