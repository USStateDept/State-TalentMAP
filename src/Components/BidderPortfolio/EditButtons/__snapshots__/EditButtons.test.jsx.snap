// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SearchResultsExportLink matches snapshot 1`] = `
<div
  className="edit-buttons-container"
>
  <button
    onClick={[Function]}
  >
    Edit
  </button>
</div>
`;
